lib/puma/app/status.rb

```diff
@@ -7,13 +7,16 @@ module App
     # can respond to.
     class Status
       OK_STATUS = '{ "status": "ok" }'.freeze
+      READ_ONLY_COMMANDS = %w[gc-stats stats].freeze
 
       # @param launcher [::Puma::Launcher]
       # @param token [String, nil] the token used for authentication
+      # @param data_only [Boolean] if true, restrict to read-only data commands
       #
-      def initialize(launcher, token = nil)
+      def initialize(launcher, token = nil, data_only = false)
```

Comment:
Please use kwargs instead of positional args here for the newly added `data_only`.

@dentarg @MSP-Greg this class doesn't have any docs regarding public/private. I'm unsure if people use it directly, you think it's okay to change `token` to also be a kwarg or is that breaking?

Reply:
> I'm unsure if people use it directly

I've got several "Don't break" bookmarks, and I don't recall any using this directly.

> you think it's okay to change token to also be a kwarg or is that breaking?

Do you really think these need to change to kwargs?  Adding an additional positional at the end, where the proceeding arg also has a default value shouldn't be a problem.  Also, I don't see the arg list growing.  Granted, kwargs are 'identifying', it not like the init code is used frequently...